lib/spack/spack/build_systems/cuda.py

```diff
@@ -144,12 +144,6 @@ def cuda_flags(arch_list):
     conflicts('%clang@4:', when='+cuda ^cuda@:9.0.176' + arch_platform)
     conflicts('%clang@5:', when='+cuda ^cuda@:9.1' + arch_platform)
     conflicts('%clang@6:', when='+cuda ^cuda@:9.2' + arch_platform)
-    conflicts('%clang@7:', when='+cuda ^cuda@10.0.130' + arch_platform)
```

Comment:
@svenevs these are the conflicts I could see happening on our system. They are just some of the conflicts @davidbeckingsale had modified with `allow-untested-versions`.

Reply:
Some versions are "unofficially" supported - these are what I wanted to allow with the new variant. We have modified CUDA installs at LLNL where the headers are modified to allow these combinations to work.

Reply:
Ahh I didn't realize we had modified headers.

Reply:
I think there's a better solution here,

- Introduce the variant for allowing unsupported. For CUDA 11+ this will require us to introduce the newly added build flag for this, and disables conflicts. (This solution won't help you).
    - This will also relate to the LLVM // CUDA work.
- Wherever and however that happens, sneak in a hack for the exact versions LLNL needs with a `NOTE` explaining this is the only exception we're going to add explaining LLNL gets special priority.

I do still think it would be better to open a separate PR for this, we can merge it without solving the CUDA 11 thing but instead open an issue to track that and reference the PR adding the variant to indicate it's already there but does not do anything yet. Basically, keep the history clean but additionally since CUDA 11 allows version checks to be suppressed, the variant will deceive people so the help string needs to explain it's not going to do what they think.

I'm not willing to sign off on this in it's current form because of the confusion it will cause ("I set allow unsupported but it doesn't work" influx of issues). But I've also stepped down from maintaining this package and think the current maintainers should have the opportunity to vet this on it's own without the other changes going on here.

Reply:
I've removed all CUDA changes from this branch including the commit history.